### PR TITLE
feat: add relation endpoints in group

### PR DESCRIPTION
**Pull request by Tlazypanda**
Ref #7597 

#### Short description of what this resolves:
Add more relationship endpoints in group feature.
/groups/ID/events
/events/ID/group

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by Tlazypanda** [comment]
@iamareebjamal while trying to hit /groups/ID/events endpoint I get `Could not build url for endpoint 'v1.event_group_detail' with values ['event_id']. Did you mean 'v1.event_topic_detail' instead?` but I am not sure how to correct this - I have defined this relationship in my group schema and have added the route under event_list, have modified the events api to take group_id param. Meanwhile the /groups/ID/relationship/events api seems to be working fine :sweat_smile:

**Comment by Tlazypanda** [comment]
@iamareebjamal my bad :facepalm: updated mr :v: can you please review now? :sweat_smile:

File: app/api/groups.py [hunk 1]
```diff
@@ -28,9 +29,16 @@ def before_post(self, args, kwargs, data):
         :return:
         """
         data['user'] = current_user.id
+        require_relationship(['event'], data)
```
Comment (iamareebjamal) [inline app/api/groups.py:32]: Not required. 

1. You should be able to create group without event 
2. Group doesn't have one event. It has events, if the schema field is named event, please change it to events

**Comment by Tlazypanda** [comment]
@iamareebjamal updated mr :sweat_smile:

File: app/api/groups.py [hunk 2]
```diff
@@ -28,9 +29,16 @@ def before_post(self, args, kwargs, data):
...
         if not current_user.is_verified:
             raise ForbiddenError({'source': ''}, 'Access Forbidden')
 
+    def after_create_object(self, group, data, view_kwargs):
+        if 'events' in data:
+            for event_id in data['events']:
+                event = safe_query(Event, 'id', event_id, 'event_id')
+                event.group_id = group.id
```
Comment (iamareebjamal) [inline app/api/groups.py:38]: This shouldn't be needed

File: app/api/users.py [hunk 3]
```diff
@@ -237,6 +238,13 @@ def before_get_object(self, view_kwargs):
             else:
                 view_kwargs['id'] = None
 
+        if view_kwargs.get('group_id') is not None:
+            group = safe_query_kwargs(Group, view_kwargs, 'group_id')
+            if group.user is not None:
+                view_kwargs['id'] = group.user.id
```
Comment (iamareebjamal) [inline app/api/users.py:244]:
```suggestion
            if group.user_id is not None:
                view_kwargs['id'] = group.user_id
```

group.user will fetch the user from DB

**Comment by iamareebjamal** [comment]
@Tlazypanda Please refrain from force pushing unless rebasing, because it makes it impossible to see incremental changes

Also, please follow the PR title guideline, feat should be prefix, not suffix and followed by a colon

**Comment by Tlazypanda** [comment]
@iamareebjamal yup definitely! :v:

**Comment by Tlazypanda** [comment]
@iamareebjamal is any other change required in this pr that i am missing ? :sweat_smile:

**Pull request merged**